sdk_extensions/zpages/src/main/java/io/opentelemetry/sdk/extensions/zpages/ZPageHandler.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.extensions.zpages;
+
+import java.io.OutputStream;
+import java.util.Map;
+
+/**
+ * The main interface for all zPages. All zPages should implement this interface to allow the HTTP
+ * server implementation to support these pages.
+ */
+public abstract class ZPageHandler {
```

Comment:
Just an aside, I won't ever start a `ZPageServer` in my apps :) And will register this directly into my preferred web server, such as Armeria.

I don't see anything in the design that prevents it and it looks good, but wanted to make sure it's on people's minds :) The only point was about the `BufferedWriter` which I wouldn't want to use in conjunction with a different server framework.

Reply:
It does make me think that we should create a Servlet that does all this stuff, so it would be super easy to plug in to any standard JEE server.